openmetadata-service/src/main/java/org/openmetadata/service/pipelineServiceClient/WorkflowConfigBuilder.java

```diff
@@ -0,0 +1,120 @@
+import org.openmetadata.service.util.EntityUtil;
+
+public class WorkflowConfigBuilder {
+  public static OpenMetadataWorkflowConfig buildOMWorkflowConfig(IngestionPipeline ingestionPipeline)
+      throws IOException {
+
+    OpenMetadataWorkflowConfig workflowConfig;
+
+    switch (ingestionPipeline.getPipelineType()) {
+      case METADATA:
+        workflowConfig = buildMetadataWorkflowConfig(ingestionPipeline);
+        break;
+      case USAGE:
+        workflowConfig = buildUsageWorkflowConfig(ingestionPipeline);
+        break;
+      case LINEAGE:
+        workflowConfig = buildLineageWorkflowConfig(ingestionPipeline);
+        break;
+      case PROFILER:
+        workflowConfig = buildProfilerWorkflowConfig(ingestionPipeline);
+        break;
+      case TEST_SUITE:
+        workflowConfig = buildTestSuiteWorkflowConfig(ingestionPipeline);
+        break;
+      case DATA_INSIGHT:
+        workflowConfig = buildDataInsightWorkflowConfig(ingestionPipeline);
+        break;
+      default:
+        throw new IllegalArgumentException("Not implemented pipeline type: " + ingestionPipeline.getPipelineType());
+    }
+    return workflowConfig;
+  }
+
+  public static Sink buildDefaultSink() {
+    Sink sink = new Sink();
+    sink.setType("metadata-rest");
+    sink.setConfig(new ComponentConfig());
+    return sink;
+  }
+
+  public static WorkflowConfig buildDefaultWorkflowConfig(IngestionPipeline ingestionPipeline) {
+    WorkflowConfig workflowConfig = new WorkflowConfig();
+    workflowConfig.setLoggerLevel(
+        ingestionPipeline.getLoggerLevel() != null ? ingestionPipeline.getLoggerLevel() : LogLevels.INFO);
+    workflowConfig.setOpenMetadataServerConfig(ingestionPipeline.getOpenMetadataServerConnection());
+    return workflowConfig;
+  }
+
+  public static Source buildDefaultSource(IngestionPipeline ingestionPipeline) throws IOException {
+    Source source = new Source();
+    source.setServiceName(ingestionPipeline.getService().getName());
+    source.setSourceConfig(ingestionPipeline.getSourceConfig());
+
+    ServiceEntityInterface service =
+        Entity.getEntity(ingestionPipeline.getService(), EntityUtil.Fields.EMPTY_FIELDS, Include.NON_DELETED);
+    source.setType(service.getServiceType().toString().toLowerCase());
+
+    return source;
+  }
+
+  public static OpenMetadataWorkflowConfig buildMetadataWorkflowConfig(IngestionPipeline ingestionPipeline)
+      throws IOException {
+    OpenMetadataWorkflowConfig config = new OpenMetadataWorkflowConfig();
+
+    Source source = buildDefaultSource(ingestionPipeline);
+    Sink sink = buildDefaultSink();
+    WorkflowConfig workflowConfig = buildDefaultWorkflowConfig(ingestionPipeline);
+
+    config.setSource(source);
+    config.setSink(sink);
+    config.setWorkflowConfig(workflowConfig);
+
+    return config;
+  }
+
+  public static OpenMetadataWorkflowConfig buildUsageWorkflowConfig(IngestionPipeline ingestionPipeline) {
+    return null;
```

Comment:
we have open issues to keep implementing these

Reply:
nit: IMHO we could apply a [strategy](https://refactoring.guru/design-patterns/strategy) pattern to build the `OpenMetadataWorkflowConfig` instead of overloading this class with static methods.